scale/node/views.py

```diff
@@ -142,9 +110,30 @@ def patch(self, request, node_id):
             raise Http404
 
         Node.objects.update_node(dict(request.data), node_id=node_id)
-        node = Node.objects.get_details(node_id)
-        serializer = NodeSerializer(node)
-        return Response(serializer.data, status=status.HTTP_201_CREATED,
+
+        # Fetch the basic node attributes
+        try:
+            node = Node.objects.get_details(node_id)
+        except Node.DoesNotExist:
+            raise Http404
+        serializer = self.get_serializer(node)
+        result = serializer.data
+
+        # Attempt to fetch resource usage for the node
+        resources = None
+        try:
+            sched = Scheduler.objects.get_master()
+            slave_info = mesos_api.get_slave(sched.master_hostname, sched.master_port, node.slave_id, True)
+            if slave_info and slave_info.total:
+                resources = slave_info.to_dict()['resources']
+        except:
+            logger.exception('Unable to fetch slave resource usage')
+        if resources:
+            result['resources'] = resources
+        else:
+            result['disconnected'] = True
+
```

Comment:
Since all of this stuff is identical to retrieving/processing the node detail that is in the GET method, factor both of these out into a helper method called something like _get_node_details() for the view. That way the logic to return node details is not duplicated.